### PR TITLE
feat: introduce linter

### PR DESCRIPTION
**Pull request by snqb**
We had no linting flow.

Now we do.
This PR includes:
- setup of eslint
- some rules
- fixes for auto-fixable things  

Tons of work still.

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -1,15 +1,12 @@
 {
   "parser": "@typescript-eslint/parser",
   "extends": [
-    "standard",
-    "standard-react",
```
Comment (snqb) [inline .eslintrc:5]: we had to choose between standard and airbnb.

I think we should go with airbnb. We use it in hops too.

File: .eslintrc [hunk 2]
```diff
@@ -36,7 +33,7 @@
     "import/export": 0,
     "no-unused-vars": "off",
     "@typescript-eslint/no-unused-vars": "off",
-    "@typescript-eslint/no-unused-vars-experimental": "error",
+    "@typescript-eslint/no-use-before-define": "off",
```
Comment (snqb) [inline .eslintrc:36]: turning off this rule allows us to write functions in the end of the file and etc.

I'll explain every rule with comments in next PR. Just trust me until that.



(everything added below was taken from `hopsworks` anyways

File: package.json [hunk 3]
```diff
@@ -18,12 +18,11 @@
     "start": "microbundle watch --no-compress --format modern,cjs --jsx React.createElement",
     "prepare": "run-s build",
     "prepack": "run-s build",
+    "lint": "eslint --ext .ts,.tsx ./",
+    "lint:fix": "yarn lint --fix",
     "test": "run-s test:unit test:lint test:build",
     "test:build": "run-s build",
-    "test:lint": "eslint --ext .ts,.tsx ./",
-    "test:unit": "cross-env CI=1 react-scripts test --env=jsdom",
-    "test:watch": "react-scripts test --env=jsdom",
```
Comment (snqb) [inline package.json:25]: I removed `react-scripts`, it was not used in any meaningful manner at all.
and we have no `jest` things anyways

File: package.json [hunk 4]
```diff
@@ -63,26 +62,21 @@
...
-    "eslint": "^6.8.0",
-    "eslint-config-airbnb-typescript": "^10.0.0",
-    "eslint-config-prettier": "^8.4.0",
-    "eslint-config-standard": "^14.1.0",
-    "eslint-config-standard-react": "^9.2.0",
-    "eslint-plugin-import": "^2.22.0",
-    "eslint-plugin-jsx-a11y": "^6.3.1",
-    "eslint-plugin-node": "^11.0.0",
+    "eslint": "^8.18.0",
+    "eslint-config-airbnb": "^19.0.4",
```
Comment (snqb) [inline package.json:66]: just upped them

**Review by jimdowling — APPROVED** [review]

**Review by mstrluke — APPROVED** [review]
Looks good!

**Pull request merged**